cmd/sink/filters/filters_test.go

```diff
@@ -15,13 +18,19 @@ import (
 	"gopkg.in/yaml.v3"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/kubernetes/fake"
 	"k8s.io/client-go/tools/cache"
 	toolsWatch "k8s.io/client-go/tools/watch"
 	sourcev1 "knative.dev/eventing/pkg/apis/sources/v1"
 )
 
+func TestMain(m *testing.M) {
+	globalKey = "kubearchive"
+	os.Exit(m.Run())
```

Comment:
will this prevent other test files that would run after this one will not run?

Reply:
[From the docs](https://pkg.go.dev/testing#hdr-Main):

> If TestMain returns, the test wrapper will pass the result of m.Run to os.Exit itself

I think this line is unecessary

Reply:
The linter told me it was needed, I disabled the linter for the function and removed the `os.Exit()` call.